### DISCUSSION TITLE
auto update failing with timeout

### DISCUSSION THREAD

**Discussion by drs-project** [bitfireAT/icsx5#24] [General]
after every update i get the following error.
When i update manually by pulling down it works...

![Screenshot_20220306-134316](https://user-images.githubusercontent.com/42349885/156923892-3f6296f1-9fb0-4e53-97f2-bd29d73ac19a.png)

**Comment by rfc2822** [comment]
Hi,

What do you mean with "After every update"? Sounds like a connectivity problem. Which connection type do you use?

**Reply by drs-project** [reply]
I mean the scheduled updates of the calendar. E.g. once a day or every hour.
I'm mostly on wifi, but it appears on LTE also. We have two devices showing the same behaviour.
Does the app write log somewhere with more infos?

Thanks for your help!

**Reply by rfc2822** [reply]
If you're familiar with adb, you can run [this script](https://github.com/bitfireAT/icsx5/blob/dev/scripts/adb-log.sh) and then use adb to view the logs.

Are those devices Samsung devices? I think there's a problem that the sync is initiated by the system before the network connection is really available.

I'd like to try if adding the "setRequiredNetworkType(NetworkType.CONNECTED)" constraint to the sync worker solves the problem. Where did you download ICSx⁵ from?

**Reply by drs-project** [reply]
I'm using a motorola g7 with icsx5 v2.0 from fdroid.
If you want I could test the fix...

**Reply by rfc2822** [reply]
Can you please write to play@bitfire.at so that I can send you the APK. But you will have to uninstall the F-Droid version before installing the APK.

**Reply by Elaborendum** [reply]
This has also been happening to me since a few updates ago, I'm using a Samsung device, was there any progress on this?

**Reply by rfc2822** [reply]
No progress or idea where this comes from; please post useful hints here

**Reply by drs-project** [reply]
Hi Ricki,
you already fixed it. You made a fix back in march:
`I'd like to try if adding the "setRequiredNetworkType(NetworkType.CONNECTED)" constraint to the sync worker solves the problem. `

I tested it, and you merged it into the app.

Guess, we can close this topic...

**Reply by rfc2822** [reply]
Yes but the problem unfortunately seems to persist on some devices (even for me, I also have a Samsung device). Apparently Samsung thinks an "available network connection" doesn't require DNS…

**Reply by Elaborendum** [reply]
I'm using 2.0.2 from F-Droid, I still have the issue too. However, it used to show "unable to resolve host "calendar.google.com". No address associated with hostname" when updating in the background. I recently unsubscribed and subscribed again to the Google calendars and I started getting the "timeout" error, but somehow I once again get the "unable to resolve host" error.

**Reply by rfc2822** [reply]
As I understand it, certain (mostly Samsung) devices start syncs and tell apps that "network connection is available" although it's still setting up the network/DNS. So the apps access the network when it's not ready yet, which leads to the DNS resolution error.

**Reply by sokai** [reply]
Another +1 … 🥴

Maybe this error occures while Android (13) is sleeping? (Or is Android using IPv6, instead of v4?) Because I regularly get this error, IMHO for all subscribed calendars.

I also mentioned a different message, see the screenshot.
![Screenshot_20221025_130939](https://user-images.githubusercontent.com/52245/197758272-0c3aef37-8d0f-44bb-b8e5-0b9de4b2fb02.jpg)

Thanks + best regards

**Reply by rfc2822** [reply]
I have this problem myself, Android 12, Samsung…

But no idea why

**Reply by sokai** [reply]
Hey!

I may withdraw my "+1" because today I found out that my Nextcloud server, that also has set up some of my fav remote calendars, also can't reach the server … :-|
Here are some log line:
> "message": "Subscription 1 could not be refreshed due to a network error",
>  "userAgent": "--",
>  "version": "25.0.2.3",
>  "exception": {
>    "Exception": "GuzzleHttp\\Exception\\ConnectException",
>    "Message": "cURL error 28: Operation timed out after 30001 milliseconds with 0 bytes received (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://i.cal.to/ical/XXXX/XXXXXXXXXXXXXXX/XX/XXXXXXXXXXXXXXXXXXXXX.ics",

#goodluck!

**Comment by rfc2822** [comment]
Tracked in #72

**Comment by rfc2822** [comment]
@drs-project @Elaborendum @sokai What is your sync interval? Can you please check

- whether the problem goes away when you "disable battery optimization" for the ICSx5 app in your Android battery settings,
- whether the problem goes away when you set the sync interval to less than 1 day (one time with and one time without "disabling battery optimization")?

**Reply by sokai** [reply]
Mine is currently set to `Once a day`, but plz see [my last comment](https://github.com/bitfireAT/icsx5/discussions/24#discussioncomment-4594847).
KR

**Comment by drs-project** [comment]
I'm syncing every 120min.

Battery optimization is disabled.

I haven't had any sync issues in a long time - I'm happy :-)

Am 15.12.22 um 15:54 schrieb Ricki Hirner:
